src/engine/source/builder/builders/opBuilderHelperNetInfoAddress.cpp

```diff
@@ -0,0 +1,230 @@
+/* Copyright (C) 2015-2022, Wazuh Inc.
+ * All rights reserved.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+#include "opBuilderHelperNetInfoAddress.hpp"
+
+#include <optional>
+#include <string>
+#include <variant>
+
+#include <baseHelper.hpp>
+#include <kvdb/kvdbManager.hpp>
+#include <wdb/wdb.hpp>
+
+namespace
+{
+
+// TODO: remove pre release or leave it just for testing
+constexpr std::string_view STREAM_SOCK_PATH = "/tmp/testStream.socket";
+
+enum class Name
+{
+    ADDRESS,
+    NETMASK,
+    BROADCAST
+};
+
+constexpr std::string_view getPath(Name field)
+{
+    switch (field)
+    {
+        case Name::ADDRESS: return "/address";
+        case Name::NETMASK: return "/netmask";
+        case Name::BROADCAST: return "/broadcast";
+        default: return "";
+    }
+}
+
+// when isIPv6 == true the third field will be set to 0, otherwise 1 for IPv4
+// agent <agent_id> netaddr save <scan_ID>|<name>|isIPv6=false|add[i]|netm[i]|broad[i]
+// agent 0001 netaddr save 1234|name|0|add0|netm0|broad0
+bool sysNetAddresTableFill(base::Event event,
```

Comment:
Please, if possible, try to join all the returns into a single one.

Reply:
Same as above